### PR TITLE
Meta "generator" tag in page header

### PR DESCRIPTION
**Pull request by Raf-atmire**
## References
* Fixes https://github.com/DSpace/dspace-angular/issues/747

## Description
This PR ensures that a "Generator" meta tag is set in the page header of each page in the DSpace UI.

## Instructions for Reviewers

List of changes in this PR:
* Ensures that a "Generator" meta tag is set in the page header of each page within DSpace
* Aded tests to verify this

How to test this PR:
* Setup this branch with a REST api that supports showing the dspaceVersion on the root endpoint
* Inspect the DSpace UI page and verify that a "Generator" meta tag is present in the page headers

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/app/core/data/root-data.service.ts
Comment (artlowel) [inline src/app/core/data/root-data.service.ts:46]: Any class with the `@dataService` annotation should implement at least findHref and findAllByHref, even if you use delegation. Otherwise linkService can crash at runtime. There's no type checking for that at the moment. https://github.com/DSpace/dspace-angular/issues/742 will address that

**Review by artlowel — CHANGES_REQUESTED** [review]
Thanks @Raf-atmire!

It works, but there's one problem with the dataservice

**Comment by Raf-atmire** [comment]
The dataservice has been updated to implement the findByHref and findAllByHref

**Review by artlowel — APPROVED** [review]
Thanks!

The tests are failing currently, but after those are fixed this can be merged in my opinion.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@Raf-atmire : This PR works, however the `content` value of the "generator" tag is slightly **wrong**.  As in DSpace 6.x, the Generator tag should look like this:
```
<meta name="Generator" content="DSpace 6.3">
```
For an example, see http://demo.dspace.org/xmlui/

Currently, in this PR, the Generator tag does _not_ include the word "DSpace" (which is import for us to tell if the website is a DSpace website).  So, it currently looks like:
```
<meta name="Generator" content="7.0-beta5-SNAPSHOT">
```

Keep in mind, this tag should be *hardcoded* to say "DSpace" (so do not use i18n here), as we want to discourage sites from changing this value.

Once this change is made, I'd be +1 this PR.  It works overall.

**Comment by Raf-atmire** [comment]
@tdonohue Thank you for the feedback! I do have one question though, currently the value `7.0-beta5-SNAPSHOT` is coming directly from the REST API as can be seen here: https://dspace7.4science.cloud/server/#/server/api
Should we hardcode Angular to prepend this value with "DSpace " or should we cover this in the REST API?

**Comment by tdonohue** [comment]
@Raf-atmire : excellent point.  You are correct in pointing out that the REST API should be returning `DSpace 7.0-beta5-SNAPSHOT` and not just `7.0-beta5-SNAPSHOT`.  I agree with you that this PR is fine as-is...it's the REST API return value that needs correcting.

I'll create a new ticket to solve this on the REST API...this PR is therefore good as-is.  Thanks!

**Review by tdonohue — APPROVED** [review]

**Comment by tdonohue** [comment]
@Raf-atmire : Could you rebase this on `main`?  I was waiting to merge this until *after* we merged #970 (which was just merged).  However, merging that has changed some `lint` rules slightly...so that this PR now throws some minor lint errors (if you run `yarn lint`) locally:
```
/src/app/core/data/root-data.service.spec.ts:21:7
ERROR: 21:7     semicolon    Missing semicolon

/src/app/core/data/root-data.service.ts:72:55
ERROR: 72:55    array-type   Array type using 'Array<T>' is forbidden. Use 'T[]' instead.
ERROR: 85:71    array-type   Array type using 'Array<T>' is forbidden. Use 'T[]' instead.
ERROR: 99:113   array-type   Array type using 'Array<T>' is forbidden. Use 'T[]' instead.
```

Once those lint errors are solved, this PR can be merged.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@Raf-atmire : Changing my approval to a "request for changes" as this has lint errors.  It needs rebasing on the latest main so that those lint errors can be resolved. See my previous comment.

**Review by tdonohue — APPROVED** [review]
👍 Thanks for fixing those lint issues @Atmire-Kristof & @Raf-atmire .  This looks good to me now & is ready to merge.

**Pull request merged**
